olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -477,66 +493,101 @@ client::CancellationToken AuthenticationClient::Impl::SignInClient(
     }
 
     client::OlpClient client = CreateOlpClient(settings_, boost::none);
+    using ProcessReturnHeaders = std::function<void(const olp::http::Headers&)>;
 
-    std::time_t timestamp =
-        std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+    auto call_sign_in_request = [&](std::time_t& in_timestamp,
+                                    ProcessReturnHeaders get_timestamp =
+                                        nullptr) -> SignInClientResponse {
```

Comment:
Looks like it's easier to return client::HttpResponse and do the work outside (parsing and time from headers).
Then you don't need to provide any kind of callbacks.

Reply:
In the end it can be an anonymous function.